src/bridges/erc4626/VaultBridge.sol

```diff
@@ -0,0 +1,122 @@
+        AztecTypes.AztecAsset memory,
+        uint256 _totalInputValue,
+        uint256,
+        uint64,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 _outputValueA,
+            uint256,
+            bool
+        )
+    {
+        // // ### INITIALIZATION AND SANITY CHECKS
+
+        if (_isValidPair(_outputAssetA.erc20Address, _inputAssetA.erc20Address)) {
+            _outputValueA = _enter(_outputAssetA.erc20Address, _totalInputValue);
+        } else if (_isValidPair(_inputAssetA.erc20Address, _outputAssetA.erc20Address)) {
+            _outputValueA = _exit(_inputAssetA.erc20Address, _totalInputValue);
+        } else {
+            revert ErrorLib.InvalidInput();
+        }
+    }
+
+    /**
+     * @notice Public Function used to preapprove vault pairs
+     * @param _vault address of erc4626 vault
+     * @param _token address of the vault asset
+     */
+    function approvePair(address _vault, address _token) public {
+        IERC20(_token).safeApprove(address(_vault), 0);
+        IERC20(_token).safeApprove(address(_vault), type(uint256).max);
+        IERC20(_token).safeApprove(address(ROLLUP_PROCESSOR), 0);
+        IERC20(_token).safeApprove(address(ROLLUP_PROCESSOR), type(uint256).max);
+        IERC20(_vault).approve(ROLLUP_PROCESSOR, type(uint256).max);
+    }
+
+    /**
+     * @notice Internal Function used to stake
+     * @dev This method deposits an exact amount of asset into an erc4626 vault
+     *
+     * @param _vault address of erc4626 vault
+     * @param _amount amount of an asset to be burned
+     * @return - the amount of shares earned by depositing which will equal the convert output value
+     */
+    // enter by deposit exact assets
+    function _enter(address _vault, uint256 _amount) internal returns (uint256) {
+        return IERC4626(_vault).deposit(_amount, address(this));
+    }
+
+    /**
+     * @notice Internal Function used to unstake
+     * @dev This method redeems an exact number of shares from an erc4626 vault
+     *
+     * @param _vault address of erc4626 vault
+     * @param _amount amount of shares to be redeemed
+     * @return - the amount of asset tokens withdrawn by redeeming shares
+     */
+    function _exit(address _vault, uint256 _amount) internal returns (uint256) {
+        return IERC4626(_vault).redeem(_amount, address(this), address(this));
+    }
+
+    /**
+     * @notice Function which checks whether a token is a vaild asset for a given vault
+     * @dev This method is used to check whether input and output assets are matching vault share/asset pairs
+     *
+     * @param _vault address of erc4626 vault
+     * @param _asset address of the vault asset token
+     * @return - boolean of whether an asset is a valid token for a vault
+     */
+    function _isValidPair(address _vault, address _asset) internal view returns (bool) {
```

Comment:
Please remove this function since it becomes unnecessary once you implement the `listVault(...)` function.